src/main/java/iudx/catalogue/server/database/ElasticClient.java

```diff
@@ -149,16 +151,18 @@ public void onSuccess(Response response) {
               for (int i = 0; i < results.size(); i++) {
                 JsonObject record = results.getJsonObject(i).getJsonObject(SOURCE);
                 JsonObject provider = new JsonObject();
-                String type = record.getJsonArray(TYPE)
-                        .getString(0);
-                if (type.equals("iudx:Provider")) {
+                Set<String> type = new HashSet<String>(new JsonArray().getList());
+                type = new HashSet<String>(record.getJsonArray(TYPE).getList());
+                type.retainAll(ITEM_TYPES);
+                String itemType = type.toString().replaceAll("\\[", "").replaceAll("\\]", "");
```

Comment:
Since method is same across multiple classes, try using a Util class which can be used for such functionalities